vars/buildPluginWithGradle.groovy

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env groovy
+
+/**
+ * Simple wrapper step for building a plugin with Gradle.
+ */
+def call(Map params = [:]) {
+    // Faster build and reduces IO needs
+    properties([
+        durabilityHint('PERFORMANCE_OPTIMIZED'),
+        buildDiscarder(logRotator(numToKeepStr: '5')),
+    ])
+
+    def repo = params.containsKey('repo') ? params.repo : null
+    def failFast = params.containsKey('failFast') ? params.failFast : true
+    def timeoutValue = params.containsKey('timeout') ? params.timeout : 60
+    if(timeoutValue > 180) {
+      echo "Timeout value requested was $timeoutValue, lowering to 180 to avoid Jenkins project's resource abusive consumption"
+      timeoutValue = 180
+    }
+
+    boolean publishingIncrementals = false
+    boolean archivedArtifacts = false
+    Map tasks = [failFast: failFast]
+    buildPlugin.getConfigurations(params, true).each { config ->
+        String label = config.platform
+        String jdk = config.jdk
+        String jenkinsVersion = config.jenkins
+        String javaLevel = config.javaLevel
+
+        String stageIdentifier = "${label}-${jdk}${jenkinsVersion ? '-' + jenkinsVersion : ''}"
+        
+        tasks[stageIdentifier] = {
+            node(label) {
+                timeout(timeoutValue) {
+                    // Archive artifacts once with pom declared baseline
+                    boolean doArchiveArtifacts = !jenkinsVersion && !archivedArtifacts
+                    if (doArchiveArtifacts) {
+                        archivedArtifacts = true
+                    }
+
+                    stage("Checkout (${stageIdentifier})") {
+                        infra.checkout(repo)
+                    }
+
+                    stage("Build (${stageIdentifier})") {
+                        if (javaLevel != null) {
+                            echo "WARNING: java.level is not supported in the  buildGradlePlugin(). This parmeter will be ignored"
```

Comment:
a) I suggest to quote the parameter name and show it's exact name as it has to be provided when calling `buildPluginWithGradle`. Otherwise a user might be confused and has to search for the real parameter name.
b) The referenced `buildGradlePlugin` step is not the correct one.

Reply:
```suggestion
                            echo "WARNING: javaLevel is not supported in buildPluginWithGradle(). This parameter will be ignored"
```